Samples/MedicationsTracker/MedicationsTracker/AddMedicationViewController.xib

```diff
@@ -0,0 +1,107 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<document type="com.apple.InterfaceBuilder3.CocoaTouch.XIB" version="3.0" toolsVersion="12120" systemVersion="16F73" targetRuntime="iOS.CocoaTouch" propertyAccessControl="none" useAutolayout="YES" useTraitCollections="YES" colorMatched="YES">
+    <device id="retina4_7" orientation="portrait">
+        <adaptation id="fullscreen"/>
+    </device>
+    <dependencies>
+        <deployment identifier="iOS"/>
+        <plugIn identifier="com.apple.InterfaceBuilder.IBCocoaTouchPlugin" version="12088"/>
+        <capability name="documents saved in the Xcode 8 format" minToolsVersion="8.0"/>
+    </dependencies>
+    <objects>
+        <placeholder placeholderIdentifier="IBFilesOwner" id="-1" userLabel="File's Owner" customClass="AddMedicationViewController" customModule="MedicationsTracker" customModuleProvider="target">
+            <connections>
+                <outlet property="doseAmountErrorLabel" destination="enJ-5o-n36" id="gEg-JZ-mra"/>
+                <outlet property="doseAmountField" destination="zDb-Pa-aaz" id="bpJ-XX-XPN"/>
+                <outlet property="doseUnitField" destination="pGL-ah-QHf" id="FOt-Mn-lPw"/>
+                <outlet property="medicationErrorLabel" destination="HGf-QJ-xas" id="7da-3D-XmA"/>
+                <outlet property="nameField" destination="mmL-qi-kM0" id="WPh-I6-B57"/>
+                <outlet property="nameTableView" destination="QS1-Ce-2wC" id="CX0-iv-8ci"/>
+                <outlet property="view" destination="i5M-Pr-FkT" id="sfx-zR-JGt"/>
+            </connections>
+        </placeholder>
+        <placeholder placeholderIdentifier="IBFirstResponder" id="-2" customClass="UIResponder"/>
+        <view clearsContextBeforeDrawing="NO" contentMode="scaleToFill" id="i5M-Pr-FkT">
```

Comment:
When you're ready, I'll show you about AutoLayout.  As it is now, if you run the app on a smaller phone, the fields can be too wide & be hidden by the keyboard.

Dismissing the keyboard is tricky, especially the Dosage number one because there isn't a Return key on the numeric keypad.  If you look in the Insights app code, KHRKeyboardAccessoryView adds the next/previous/done button above the keyboard.  It's handy to not have to do other weird stuff like detecting taps on the background.  You can try using as-is in Objective-C to start, everything looks like it will work except the styling it does.